src/modules/site-v2/base/static/js/scatter-histogram.js

```diff
@@ -0,0 +1,253 @@
+function buffered_extent(data, buffer, f=null) {
+  const r = d3.extent(data, f);
+  r[0] -= buffer;
+  r[1] += buffer;
+  return r;
+}
+
+
+
+/* Add a histogram along an axis of a graph.
+ *
+ * Arguments:
+ *   - d (int): The dimension of the histogram. 0 for x-axis, 1 for y-axis
+ *   - axis: A d3 scaleLinear object mapping the domain of the given axis to its position in the graph
+ *   - data: The data to plot
+ *   - target: The SVG element to create the histogram in
+ *   - config: Optional keyword arguments.
+ */
+function add_histogram_along_axis(d, axis, data, target, config) {
+
+  // Validate the dimension parameter
+  if (!(d === 0 || d === 1)) {
+    throw new RangeError(`The argument "d" must be 0 or 1, but ${d} was given.`);
+  }
+
+  // Read values from config, filling in default values when not supplied
+  const position     = config['position'] || [0, 0];
+  const graph_height = config['height']   || 60;
+  const color        = config['color']    || 'black';
+  const num_bins     = config['num_bins'] || ((axis.ticks().length - 1) * (config['bins_per_tick'] || 1));
+
+  // Get the x/y value for the provided dimension, plus the relative directions of the histogram
+  // In this case, the "width" is along the main axis, and the "height" is the direction of the individual bars
+  const [d_str, relative_width, relative_height] = [
+    ['x', 'width', 'height'],
+    ['y', 'height', 'width'],
+  ][d];
+
+  // Create a graph object for the histogram and transform it to lie on the appropriate axis
+  const graph = target.append("g")
+    .attr("transform", `translate(${ position[0] }, ${ position[1] })`);
+
+  // Create the histogram bins
+  const bins = d3.histogram()
+    .domain(axis.domain())
+    .thresholds(axis.ticks(num_bins))
+    .value(n => n[d])
+    (data);
+
+  // Compute the size of the bin with the max value
+  // This will be used to scale the histogram
+  const max_bin_amount = d3.max(bins, n => n.length);
+
+  // Linear scale from the value of a bin to its height on the page
+  const bin_val_to_height = d3.scaleLinear().domain([0, max_bin_amount]).range([0, graph_height]);
+
+  // Compute the "width" of a single bin, based on the start & end coords of the second bin
+  // I use the second bin here in case the axis truncates in between ticks (i.e. isn't "niced"),
+  // which would make the first bin a bit smaller
+  const bin_width = Math.abs(axis(bins[1].x1) - axis(bins[1].x0)) - 1;
+
+  // Fill each bar with the data from the appropriate bin,
+  // and translate along the primary axis so they all sit next to one another
+  const bar = graph.selectAll('.bar')
+    .data(bins)
+    .enter().append('g')
+    .attr('class', 'bar')
+    .attr('transform', function(n) {
+
+      // D3 draws from top left, so along the x-axis,
+      // offset the y-position of the bar based on the white space that should be above it
+      if (d == 0) return `translate(${ axis(n.x0) }, ${ bin_val_to_height(max_bin_amount - n.length) })`;
```

Comment:
consider using triple equals `===` on these comparisons to avoid type coercion side effects (eg: `false == 0` is `true`)

Reply:
Fixed in 7651b21e7c0ec4105403c84ddef0a8bb07ff348c